universal-application-tool-0.0.1/app/services/program/ProgramServiceImpl.java

```diff
@@ -414,12 +414,19 @@ public ProgramDefinition setBlockPredicate(
 
```

Comment:
suggestion: We should add the `@Nullable` annotation to indicate `predicate` can be null, like this:

```
setBlockPredicate(long programId, long blockDefinitionId, @Nullable PredicateDefinition predicate)
```

(I can't make a comment on that line in Github)

Reply:
Ohh nice! Will do.